pkg/registry/core/service/rest.go

```diff
@@ -446,6 +457,17 @@ func contains(haystack []int, needle int) bool {
 	return false
 }
 
+// This is O(N), but we expect serviceNodePorts to be small;
+// so small that we expect a linear search to be faster
+func containsNodePort(serviceNodePorts []ServiceNodePort, serviceNodePort ServiceNodePort) bool {
+	for _, snp := range serviceNodePorts {
+		if reflect.DeepEqual(snp, serviceNodePort) {
```

Comment:
should be able to do `==` for comparing structs containing only primitives, right?

Reply:
Updated. Thanks!